app/services/teacher_training_public_api/sync_all_providers_and_courses.rb

```diff
@@ -1,15 +1,19 @@
 module TeacherTrainingPublicAPI
   class SyncAllProvidersAndCourses
-    def self.call(recruitment_cycle_year: ::RecruitmentCycle.current_year)
+    def self.call(recruitment_cycle_year: ::RecruitmentCycle.current_year, incremental_sync: true)
       is_last_page = false
       page_number = 0
       until is_last_page
         page_number += 1
-        response = TeacherTrainingPublicAPI::Provider
+
+        scope = TeacherTrainingPublicAPI::Provider
           .where(year: recruitment_cycle_year)
           .paginate(page: page_number, per_page: 500)
-          .all
-        sync_providers(response, recruitment_cycle_year)
+        scope = scope.where(updated_since: TeacherTrainingPublicAPI::SyncCheck.updated_since) if incremental_sync
```

Comment:
So i have one question about this. This basically means that we get all the records updated in the last 2ish hours right?

Would it be better if we used this?

![image](https://user-images.githubusercontent.com/42515961/118622440-3c1b7100-b7bf-11eb-8b26-4b47a843f79e.png)


I'm just thinking that at some point the sync might fail a few times and we would lose data until a full sync happens 3 days later.

Reply:
Yeah my thoughts exactly. We may need to take into account that the incremental sync fails so `TeacherTrainingPublicAPI::SyncCheck.updated_since` offsets `last_sync` by two hours. Can increase it if you like though to be safe?

`(Time.zone.parse(last_sync) - 2.hours)`

https://github.com/DFE-Digital/apply-for-teacher-training/pull/4712/files#diff-86a2f62cd0d36bb071a44016c9f545566bce203ff84184ddbb07031f374e6dfbR17